framework/include/utils/MortarUtils.h

```diff
@@ -274,14 +283,62 @@ loopOverMortarSegments(const Iterators & secondary_elems_to_mortar_segments,
       fe_problem.reinitMaterialsFace(secondary_ip->subdomain_id(),
                                      /*tid=*/0,
                                      /*swap_stateful=*/false,
-                                     /*execute_stateful=*/false);
+                                     &secondary_ip_mats);
       fe_problem.reinitMaterialsBoundary(
-          secondary_boundary_id, /*tid=*/0, /*swap_stateful=*/false, /*execute_stateful=*/false);
+          secondary_boundary_id, /*tid=*/0, /*swap_stateful=*/false, &secondary_boundary_mats);
 
       act();
 
     } // End loop over msm segments on secondary face elem
   }   // End loop over (active) secondary elems
 }
+
```

Comment:
If you touch the code again, maybe adding some comments for this method would be helpful in the future.